### ISSUE TITLE
enable_identity_service ignores false once true

### ISSUE THREAD

**Issue by bjhshadow** [terraform-google-modules/terraform-google-kubernetes-engine#2129]
### TL;DR

The input _enable_identity_service_ will enable the GKE (anthos) identity service if set to true.   Once enabled, the identity service option will not trigger once set to false 

### Expected behavior

The boolean input option enable_identity_service should enable when **true** and disable when **false**.  

### Observed behavior

During cluster create, the _enable_identity_service_ option checks for boolean value and proceeds as expected to either:

1. if true, enable the GKE identity service
2. if false, not enable the service or
3. if not set, not enable the service as the default is also false

Once enabled, modifying _enable_identity_service_ to false creates **no action or change** when planning or applying and the Identity Service remains active on the GKE cluster.

### Terraform Configuration

```hcl
module "gke" {
  source   = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant"
  version  = "31.1.0"
  ...

  enable_identity_service = false
  ...
}
```

### Terraform Version

```sh
$ terraform version
Terraform v1.5.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v5.44.1
+ provider registry.terraform.io/hashicorp/google-beta v5.44.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.32.0
+ provider registry.terraform.io/hashicorp/random v3.6.3
+ provider registry.terraform.io/hashicorp/time v0.12.1
```

**Comment by bjhshadow** [comment]
A cluster which had identity service enabled and running that subsequently was modified outside of terraform to with no-enable-identity-service no longer inspects current state for that configuration.

Step (0) The original Terraform Configuration

module "gke" {
  source   = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant"
  version  = "31.1.0"
  ...

  enable_identity_service = true
  ...
}

Step (1): Describe Cluster

gcloud container clusters describe cluster33 --region us-central1 | grep -A 2 -i identityServiceConfig
    identity_service_config {   
      enabled = true    
     }  

$ terraform state show 'module.gke-enterprise-clusters.module.gke["cluster33"].google_container_cluster.primary' | grep -A 2 identity_service_config
    identity_service_config {
        enabled = true
    }

Step (2): Manually disable identity service

$ gcloud container clusters update cluster33 --no-enable-identity-service --
region us-central1
Updating cluster33...done.

s$ gcloud container clusters describe cluster33 --region us-central1 | grep -A 1 -i identityServiceConfig
identityServiceConfig: {}
initialClusterVersion: 1.30.3-gke.1969001

Step (3) Rerun terraform

$ terraform state plan
...
No changes. Your infrastructure matches the configuration.

$ terraform state show 'module.gke-enterprise-clusters.module.gke["gke-np-tsbx-33"].google_container_cluster.primary' | grep -A 2 identity_service_config
    identity_service_config {
        enabled = true
    }
t187106@C510829:~/gke-enterprise-technical-sandbox/gke-clusters$ gcloud container clusters describe gke-np-tsbx-33 --region us-central1 | grep -A 1 -i identityServiceConfig
identityServiceConfig: {}
initialClusterVersion: 1.30.3-gke.1969001

**Comment by apeabody** [comment]
> I did a quick test with the latest version of this module and the provider version you're using (`5.44.1`) and created a cluster with that setting set to `false`.
> 
> If I then change the setting to `true`, I see this in the plan
> 
> ```
>   # module.gke.google_container_cluster.primary will be updated in-place
>   ~ resource "google_container_cluster" "primary" {
>         id                                       = "projects/xx/locations/us-central1/clusters/test-cluster"
>         name                                     = "test-cluster"
>         # (30 unchanged attributes hidden)
> 
>       + identity_service_config {
>           + enabled = true
>         }
> 
>         # (23 unchanged blocks hidden)
>     }
> ```
> 
> and the cluster takes a while to update and then updates it:
> 
> ```
> % gcloud container clusters describe test-cluster --location us-central1 --project xxx | yq .identityServiceConfig
> enabled: true
> ```
> 
> However, going in the other direction, as described, there's no diff in the plan whether removing the block entirely from the config, or setting it explicitly to `false`
> 
> So possible that the code needs to be updated to something like this, but then the default might need to be updated to `null`?
> 
> ```hcl
>     for_each = var.enable_identity_service !=null ? [var.enable_identity_service] : []
> ```
> 
> or else the block should just be changed to not be a dynamic block at all and always be explicitly `true` or `false`. Thoughts, @apeabody?

Hi @wyardley - I'd prefer to keep the dynamic block, even if the default remains `false` it is possible to set `enable_identity_service = null`.  So yeah, I think your suggest code change above may resolve the issue.

**Comment by wyardley** [comment]
It fixed it for me locally in a simple test.i can throw up a PR

Will terraform still allow the bool to be set to null if that’s not its default?
